pandas/io/parsers.py

```diff
@@ -393,11 +393,15 @@ def _read(filepath_or_buffer, kwds):
         raise NotImplementedError("'nrows' and 'chunksize' cannot be used"
                                   " together yet.")
     elif nrows is not None:
-        return parser.read(nrows)
+        data = parser.read(nrows)
```

Comment:
if you do

```
try:
    return parser.read(nrows)
finally:
    parse.close()
```

I think you can eliminate the ``.close()`` calls where you raise on an exception (also needs changing at L402).
and slightly more idiomatic.